### ISSUE TITLE
Try to open missing PDF backgrounds from current directory

### ISSUE THREAD

**Issue by timholl** [xournalpp/xournalpp#4202]
**Is your feature request related to a problem? Please describe.**

Currently, the Path to the PDF background(s) are stored as *absolute* paths within the `xopp`-File, and there is no way to change that (at least I can not see it).
   ```xml
   <background type="pdf" domain="attach" filename="/path/to/file.pdf" pageno="1ll"/>
   ```
Therefore, when I move a pair of PDF-and-xopp-Files next to each other to another location, it reports the background as missing.

This problem has been reported here, e.g. #2736 #2638 , but no solution has been found yet.

**Describe the solution you'd like**
I am suggesting the functionality of falling back to search the missing background with its original filename in the directory that the `xopp`-file is located in.

**Describe alternatives you've considered**

The `DocumentLoader` is able to read from relative paths, so an alternative (without breaking current behaviour/already stored files) would be to let the user store relative paths in the first place. However, this would not be helpful when opening old `xopp`-Files, which have been moved.

**Additional context**

Here is a quick-and-dirty patch I am currently using and I can confirm working.

```patch
Index: xournalpp-1.1.1/src/control/xojfile/LoadHandler.cpp
===================================================================
--- xournalpp-1.1.1.orig/src/control/xojfile/LoadHandler.cpp
+++ xournalpp-1.1.1/src/control/xojfile/LoadHandler.cpp
@@ -459,6 +459,17 @@ void LoadHandler::parseBgPdf() {
 
         this->pdfFilenameParsed = true;
 
+        /*
+         * Try to open PDF backgrounds, which can not be found on their original location, from the XournalFile's directory.
+         */
+        if (!fs::is_regular_file(pdfFilename)) {
+            fs::path fallback = xournalFilepath.remove_filename() / pdfFilename.filename();
+            if (fs::is_regular_file(fallback)) {
+                g_warning("LoadHandler::parseBgPdf: Using replaced background, as original can not be found.");
+                pdfFilename = fallback;
+            }
+        }
+
         if (fs::is_regular_file(pdfFilename)) {
             doc.readPdf(pdfFilename, false, attachToDocument);
             if (!doc.getLastErrorMsg().empty()) {
```

Of course, one could also add a dialogue such as

> "The background X can not be found, but there is a file with the same name in the current directory (X'), do you want to use it instead?"

to allow the user more control.

I am open to submit a PR and to add/improve the functionality. Also, feel free to close this issue if you think this behaviour should not be included.

**Comment by Malex14** [comment]
This is already implemented by my PR #4165, that has been merged a few weeks ago. 

> If there is a PDF file with the same name as the missing PDF in the directory where the journal file is saved, the dialog now proposes that file to the user. If not the dialog stays as before.
> 
> This is especially useful when moving the journal file and the background to a new location. Or in my case where I switch between Linux and Windows a lot, the background can be automatically found even tho Linux and Windows are using a different path structure
>

**Comment by timholl** [comment]
Wonderful!
Sorry, had not tested with the `dev` version.

Since you are the PR author, thank you! That makes my life a bit easier :)

Closing this issue now.
